tasks/kernel_matrix_testing/README.md

```diff
@@ -1,425 +1,564 @@
-
-# If you intend to use KMT *only* for managing remote VMs
-dda inv -e kmt.init --remote-setup-only
 ```
 
-The `--images` parameter is required unless `--remote-setup-only` or `--all-images` is specified. You can use `dda inv -e kmt.ls` to see available images.
+**2. Create and Configure a Stack**
 
-This command will also ask you for the default SSH key to use, so it does not need to be provided every time. You can configure the SSH key again at any time running `dda inv -e kmt.config-ssh-key`. See more details below
+A stack is a collection of VMs. This command will create a new stack and configure it to run one local Ubuntu 22.04 VM.
 
-#### SSH key configuration
+```bash
+# --init-stack creates the stack if it does not exist
+# If --stack is not provided, a name is generated from your git branch
+dda inv -e kmt.gen-config --vms=ubuntu22-local-distro --init-stack --stack=my-first-stack
+```
 
-In order to create EC2 instances, we need to specify which SSH key will be provisioned on them. As that key is usually always the same, we ask the user to provide it once and store it in the configuration file. This way, the user does not need to provide it every time they launch a stack (although it's possible to change it by using the argument `--ssh-key` with KMT tasks).
+**3. Launch the Stack**
 
-The configuration wizards will ask for the SSH key to use. There are three methods supported:
+This will start the VM(s) in your stack.
 
-- Keys stored in `.ssh`. The wizard will automatically look for key files present there, and you will be able to choose one of them.
-- Keys stored in a SSH agent (main use: 1Password SSH agent). The wizard will look for agent keys. Note that with this method we don't know the path to the private key file, which means that SSH configurations will not specify the key to use. This might be problematic if you have a lot of keys in the agent, as SSH will try all of them and it might reject your connection before finding the correct one. In that case you should download the public key from 1Password and configure it manually. [See more details in 1Password docs](https://developer.1password.com/docs/ssh/agent/advanced/).
-- Manual input. If you have a key that is not in `.ssh` or in the agent, you can provide the path to the private key file and key name. Note that **this method performs no validation**.
+```bash
+dda inv -e kmt.launch-stack --stack=my-first-stack
+```
+> Since this is a local VM, you may be prompted for your password to run some commands with `sudo`.
 
-In all cases, you will be able to input a key name for AWS, in case the keypair you stored in AWS has a different name than the name present in the key file.
+**4. Check Status**
 
-### Create stack
+You can view the running VMs and their IP addresses.
 
 ```bash
-dda inv -e kmt.create-stack --stack=demo-stack
+dda inv -e kmt.status --stack=my-first-stack
 ```
 
-### Configure stack
-
-We will configure the stack to launch
+**5. Connect to the VM**
```

Comment:
I think this should not be listed in the **Quick Start** section, since this is more of a "power user" feature.

The normal way to interact with the VMs is to use `inv kmt.test` for running tests against the VMs.

Reply:
Good Point. I agree that kmt.test is the standard way to interact with the VMs, which is why I've highlighted it as the "primary way" in that section.

I chose to include the SSH instructions as an optional, secondary step because debugging is a very common part of the development workflow. My goal for the Quick Start is to give developers the essential information to be productive right away, and that includes knowing how to inspect the environment if a test fails.

Since the section is clearly marked as optional, it shouldn't distract from the main workflow.